examples/gno.land/p/demo/int256/arithmetic.gno

```diff
@@ -0,0 +1,187 @@
+package int256
+
+import (
+	"gno.land/p/demo/uint256"
+)
+
+func (z *Int) Add(x, y *Int) *Int {
+	z.initiateAbs()
+
+	neg := x.neg
+
+	if x.neg == y.neg {
+		// x + y == x + y
+		// (-x) + (-y) == -(x + y)
+		z.abs = z.abs.Add(x.abs, y.abs)
+	} else {
+		// x + (-y) == x - y == -(y - x)
+		// (-x) + y == y - x == -(x - y)
+		if x.abs.Cmp(y.abs) >= 0 {
+			z.abs = z.abs.Sub(x.abs, y.abs)
+		} else {
+			neg = !neg
+			z.abs = z.abs.Sub(y.abs, x.abs)
+		}
+	}
+	z.neg = neg // 0 has no sign
+	return z
+}
+
+// AddUint256 set z to the sum x + y, where y is a uint256, and returns z
+func (z *Int) AddUint256(x *Int, y *uint256.Uint) *Int {
+	if x.neg {
+		if x.abs.Gt(y) {
+			z.abs.Sub(x.abs, y)
+			z.neg = true
+		} else {
+			z.abs.Sub(y, x.abs)
+			z.neg = false
+		}
+	} else {
+		z.abs.Add(x.abs, y)
+		z.neg = false
+	}
+	return z
+}
+
+// Sets z to the sum x + y, where z and x are uint256s and y is an int256.
+func AddDelta(z, x *uint256.Uint, y *Int) {
+	if y.neg {
+		z.Sub(x, y.abs)
+	} else {
+		z.Add(x, y.abs)
+	}
+}
+
+// Sets z to the sum x + y, where z and x are uint256s and y is an int256.
+func AddDeltaOverflow(z, x *uint256.Uint, y *Int) bool {
+	var overflow bool
+	if y.neg {
+		_, overflow = z.SubOverflow(x, y.abs)
+	} else {
+		_, overflow = z.AddOverflow(x, y.abs)
+	}
+	return overflow
+}
+
```

Comment:
Why are these two functions, and not member methods like the rest? Is this the ported library API?

Reply:
Origin functions were [defined](https://github.com/mempooler/int256/blob/7a148ada2e2a5c990ebbe412dc68db25acf2eac8/int256.go#L551-L569) without receiver